compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -420,6 +420,15 @@ class ExpressionCodegen(
     }
 
     override fun visitFieldAccess(expression: IrFieldAccessExpression, data: BlockInfo): PromisedValue {
+        classCodegen.fieldConstantValue(expression.symbol.owner)?.let {
```

Comment:
@pyos Is it possible to move this part to ConstLowering? In such case additional optimization would be applied during lowers phase, e.g. foldConstantLoweringPhase

Reply:
Yes, I'm actually looking at this right now.

Reply:
@max-kammerer I've moved inlining of Java constants into ConstLowering and added an assertion to codegen to ensure that nothing has been missed, could you have a look, please?